library/src/commonMain/kotlin/com/badoo/reaktive/observable/Throttle.kt

```diff
@@ -0,0 +1,20 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.utils.uptimeMillis
+
+fun <T> Observable<T>.throttle(window: Long): Observable<T> =
+    observable { observer ->
+        subscribeSafe(
+            object : ObservableObserver<T> by observer {
+                private var lastTime = 0L
+
+                override fun onNext(value: T) {
+                    val time = uptimeMillis
```

Comment:
Would be easier for testing if we somehow avoided the singleton style of checking time.
If this could perhaps be passed inside an internal constructor or something.

Reply:
I thought about it, but there is no constructor here at all. We probably can mock "static" time provider in tests. Or we can do same with RxJava2: use scheduler to open gate again when timeout passed. Then we can actually remove uptimeMillis at all as this is the only usage at the moment.

Reply:
Makes sense to remove the uptimeMillis.

Reply:
I would like to discuss this one, we can fix it separately later on.